azure-servicebus/src/main/java/com/microsoft/azure/servicebus/IMessage.java

```diff
@@ -9,49 +9,74 @@
 import java.util.UUID;
 
 /**
- * Represents a message that is exchanged between Azure Service Bus and clients.
+ * Represents the message that is exchanged between Azure Service Bus and its clients.
  *
  * @since 1.0
+ * @see <a href="https://docs.microsoft.com/azure/service-bus-messaging/service-bus-messages-payloads">Messages, Payloads, and Serialization</a>
  */
 public interface IMessage {
+
     /**
      * Gets the number of the times this message was delivered to clients.
+     * 
+     * The count is incremented when a message lock expires, or the message is 
+     * explicitly abandoned by the receiver. This property is read-only.
```

Comment:
>This property is read-only. [](start = 45, length = 27)

Nit: Not a property per se. In java, this is only a getter. Absence of setter means readonly.